File: MVVMQuize/app/src/main/java/ru/sport/mvvmquize/QuizRepository.java
package ru.sport.mvvmquize;

import java.util.List;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class QuizRepository {
    private QuizApi quizApi;

    public QuizRepository() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("http://159.69.90.204/api/MVVMQuize/") // Замените на базовый URL вашего сервера
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        quizApi = retrofit.create(QuizApi.class);
    }

    public Call<List<Question>> getQuestions() {
        return quizApi.getQuestions();
    }
}

File: KTprila2/app/src/main/java/com/sport/ktprila/QuizRepository.kt
package com.sport.ktprila

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class QuizRepository {
    private val api: QuizApi

    init {
        val retrofit = Retrofit.Builder()
            .baseUrl("http://159.69.90.204/api/Ktprila/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        api = retrofit.create(QuizApi::class.java)
    }

    suspend fun getQuizQuestions(): List<Question> {
        return try {
            val response = api.getQuizQuestions()
            if (response.isSuccessful) {
                response.body()?.questions ?: emptyList()
            } else {
                emptyList()
            }
        } catch (e: Exception) {
            emptyList()
        }
    }
}
